include/graaflib/algorithm/shortest_path.tpp

```diff
@@ -111,4 +111,47 @@ std::optional<graph_path<WEIGHT_T>> dijkstra_shortest_path(
   return reconstruct_path(start_vertex, end_vertex, vertex_info);
 }
 
+template <typename V, typename E, graph_type T, typename WEIGHT_T>
+[[nodiscard]] std::unordered_map<vertex_id_t, graph_path<WEIGHT_T>>
+dijkstra_shortest_paths(const graph<V, E, T>& graph,
+                        vertex_id_t source_vertex) {
+  std::unordered_map<vertex_id_t, graph_path<WEIGHT_T>> shortest_paths;
+
+  std::priority_queue<std::pair<WEIGHT_T, vertex_id_t>,
+                      std::vector<std::pair<WEIGHT_T, vertex_id_t>>,
+                      std::greater<>>
+      to_explore;
+
+  shortest_paths[source_vertex].total_weight = 0;
+  shortest_paths[source_vertex].vertices.push_back(source_vertex);
+  to_explore.emplace(0, source_vertex);
+
+  while (!to_explore.empty()) {
+    WEIGHT_T curr_distance = to_explore.top().first;
+    vertex_id_t curr_vertex_id = to_explore.top().second;
+    to_explore.pop();
+
+    if (shortest_paths.find(curr_vertex_id) != shortest_paths.end() &&
+        curr_distance > shortest_paths[curr_vertex_id].total_weight) {
+      continue;
+    }
+
+    for (const auto& neighbor : graph.get_neighbors(curr_vertex_id)) {
```

Comment:
[very minor]: I just noticed that this was also in some other places in the codebase, but here we are taking a reference to a small trivial type (`vertex_id_t` := `std::size_t`). 

More idiomatic would be to pass by value instead of pass by reference (which is equivalent to passing the value of the address); the `std::size_t` is 8 bytes (on a 64-bit machine), which is the same size of an address on most machines. So passing by value saves us an indirection when we use it.
```suggestion

    for (const auto neighbor : graph.get_neighbors(curr_vertex_id)) {
```